cli/src/description_util.rs

```diff
@@ -327,6 +330,53 @@ pub fn join_message_paragraphs(paragraphs: &[String]) -> String {
         .join("\n")
 }
 
+/// Add the footer entries in the last paragraph of the description
+///
+/// It just lets the description untouched if they are already there.
+/// It also let the description untouched when the description is empty,
+/// because at least one line is required in the description to define
+/// a footer paragraph.
+pub fn add_footer_entries(
+    ui: &Ui,
+    tx: &WorkspaceCommandTransaction,
+    commit: &Commit,
+    description: &str,
+    footer_templates: &[String],
+) -> Result<String, CommandError> {
+    if description.is_empty() || footer_templates.is_empty() {
+        return Ok(description.to_owned());
+    }
+    let mut new_description = description.to_owned();
+    let mut footer_entries = parse_description_footer(description);
+    if footer_entries.is_empty() {
+        // create a new paragraph for the footer
+        new_description.push('\n');
+    }
+    for template_text in footer_templates {
+        let template = tx.parse_commit_template(ui, template_text)?;
+        let mut output = Vec::new();
+        template
+            .format(commit, &mut PlainTextFormatter::new(&mut output))
+            .expect("write() to vec backed formatter should never fail");
+        // Template output is usually UTF-8, but it can contain file content.
+        let mut footer_line = output.into_string_lossy();
```

Comment:
I would suggest making this an error to start with. We can relax it later if we hear a good reason to put lossily converted non-utf8 in a trailer